gc/base/standard/Scavenger.cpp

```diff
@@ -5081,20 +5091,23 @@ MM_Scavenger::handleSurvivorCopyScanCache(MM_EnvironmentStandard *currentEnv, MM
 			 *    it will block since a) it's forced to reacquire VM access through slow path and b) caller holds thread public mutex what prevents reacquire
 			 */
 			targetEnv->_survivorCopyScanCache = NULL;
-			Assert_MM_true(cache->flags & OMR_SCAVENGER_CACHE_TYPE_COPY);
-			/* either we are explicitly instructed to flush, or we are observing suboptimal parallelism in background threads */
-			if (flushCaches || (_waitingCount > 0)) {
+			bool remainderCreated = clearCache(targetEnv, cache);
+			if (flushCaches || (_waitingCount > 0) || !remainderCreated) {
+				/* Either we are explicitly instructed to flush, or we are observing suboptimal parallelism in background threads,
+				 * or no free space in the cache -> don't deactivate, but just push it for scanning  */
+				Assert_MM_true(cache->flags & OMR_SCAVENGER_CACHE_TYPE_COPY);
 				cache->flags &= ~OMR_SCAVENGER_CACHE_TYPE_COPY;
 #if defined(J9MODRON_TGC_PARALLEL_STATISTICS)
 				targetEnv->_scavengerStats._releaseScanListCount += 1;
 #endif /* J9MODRON_TGC_PARALLEL_STATISTICS */
-				clearCache(targetEnv, cache);
 				addCacheEntryToScanListAndNotify(targetEnv, cache);
 			} else {
 				/* If it's an intermediate release (mutator threads releasing VM access in a middle of concurrent cycle), and high parallelism,
 				 * we'll keep the cache around, but tag it inactive, so it can be safely flushed at a later point (toward the end of concurrent phase)
-				 * by another thread.
+				 * by another thread. Setting inactive cache flag, just for debugging purposes.
 				 */
+				Assert_MM_false(cache->flags & OMR_SCAVENGER_CACHE_TYPE_INACTIVE);
```

Comment:
Would you please use Assert_MM_true instead for flags check? (here and everywhere is applicable):
```suggestion
				Assert_MM_true(0 == (cache->flags & OMR_SCAVENGER_CACHE_TYPE_INACTIVE));
```